api/src/main/java/bio/terra/tanagra/underlay/entitygroup/CriteriaOccurrence.java

```diff
@@ -0,0 +1,199 @@
+    Entity occurrenceEntity = deserializeEntity(serialized, OCCURRENCE_ENTITY_NAME, entities);
+    Entity primaryEntity = entities.get(primaryEntityName);
+
+    Map<String, Relationship> relationships =
+        Map.of(
+            OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME,
+                new Relationship(
+                    OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME, occurrenceEntity, criteriaEntity),
+            OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME,
+                new Relationship(
+                    OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME, occurrenceEntity, primaryEntity));
+    Map<String, AuxiliaryData> auxiliaryData =
+        Map.of(
+            CRITERIA_PRIMARY_ROLLUP_COUNT_AUXILIARY_DATA_NAME,
+            CRITERIA_PRIMARY_ROLLUP_COUNT_AUXILIARY_DATA);
+
+    EntityGroupMapping sourceDataMapping =
+        EntityGroupMapping.fromSerializedForSourceData(
+            serialized.getSourceDataMapping(), dataPointers, relationships, auxiliaryData);
+    EntityGroupMapping indexDataMapping =
+        EntityGroupMapping.fromSerializedForIndexData(
+            serialized.getIndexDataMapping(), dataPointers, relationships, auxiliaryData);
+
+    Builder builder = new Builder();
+    builder
+        .name(serialized.getName())
+        .relationships(relationships)
+        .auxiliaryData(auxiliaryData)
+        .sourceDataMapping(sourceDataMapping)
+        .indexDataMapping(indexDataMapping);
+    return builder
+        .criteriaEntity(criteriaEntity)
+        .occurrenceEntity(occurrenceEntity)
+        .primaryEntity(primaryEntity)
+        .build();
+  }
+
+  @Override
+  public EntityGroup.Type getType() {
+    return Type.CRITERIA_OCCURRENCE;
+  }
+
+  @Override
+  public Map<String, Entity> getEntities() {
+    return ImmutableMap.of(
+        CRITERIA_ENTITY_NAME, criteriaEntity, OCCURRENCE_ENTITY_NAME, occurrenceEntity);
+  }
+
+  @Override
+  public List<WorkflowCommand> getIndexingCommands() {
+    return List.of(PrecomputeCounts.forEntityGroup(this));
+  }
+
+  public Entity getCriteriaEntity() {
+    return criteriaEntity;
+  }
+
+  public Entity getOccurrenceEntity() {
+    return occurrenceEntity;
+  }
+
+  public Entity getPrimaryEntity() {
+    return primaryEntity;
+  }
+
+  public RelationshipMapping getOccurrenceToCriteriaRelationshipMapping() {
+    return sourceDataMapping
+        .getRelationshipMappings()
+        .get(OCCURRENCE_TO_CRITERIA_RELATIONSHIP_NAME);
+  }
+
+  public RelationshipMapping getOccurrenceToPrimaryRelationshipMapping() {
+    return sourceDataMapping.getRelationshipMappings().get(OCCURRENCE_TO_PRIMARY_RELATIONSHIP_NAME);
+  }
+
+  public AuxiliaryDataMapping getCriteriaPrimaryRollupCountAuxiliaryDataMapping() {
+    return indexDataMapping
+        .getAuxiliaryDataMappings()
+        .get(CRITERIA_PRIMARY_ROLLUP_COUNT_AUXILIARY_DATA_NAME);
+  }
+
+  public Query queryCriteriaPrimaryPairs(String criteriaIdAlias, String primaryIdAlias) {
```

Comment:
Do the strings passed here have any purpose? They seem to arbitrary in the one place this is called.

Reply:
The reason I included this is because the indexing workflows expect input SQL queries that have specific column names. I'd like to change that, so the caller can override the column names (and in some places you can already) but didn't want to tackle it in this already very big PR. Since Tanagra won't use these specific column names by default, I added a way to alias them to whatever the indexing workflow needs.